core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -66,6 +66,21 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
       Future.successful(snapshotRef)
     }
 
+  def cloneSnapshotByReference(sourceWorkspaceId: UUID,
+                               targetWorkspaceId: UUID,
+                               snapshot: DataRepoSnapshotResource
+  ): DataRepoSnapshotResource = {
+    if (!workspaceStubExists(targetWorkspaceId, ctx)) {
```

Comment:
Since this method is currently only used internally and doesn't care about the return value, I haven't wrapped it in a `Future`. I am happy to adjust upon request.